File: java/magic_8_ball/Magic8.java
// https://www.geeksforgeeks.org/generating-random-numbers-in-java/
import java.util.Random;
public class Magic8 {
	public static void main(String[] args) {
		System.out.println("Hello Stranger,");
		System.out.println("I see you want to find the answer to your question...");
		System.out.println("... but 100 and 100 questions swirl in your head. ");
		System.out.println("Dont't be shy, choose one: ");

		// https://www.geeksforgeeks.org/ways-to-read-input-from-console-in-java/
		// get user input:
    String myQuestion = "Hidden";
    try {
		myQuestion = System.console().readLine();
    } catch(Exception e) {
        System.out.println("Oups... this environment does not support user input. Please think a question.");
    }
		System.out.println("Hmm... Give me a minute...");

		// generate random int (0-19):
    Random rand = new Random(); 
		int replyIndex = rand.nextInt(20);

		// reply dataset:
		String[] replies = {
			"It is certain.",
			"It is decidedly so.",
			"Without a doubt.",
			"Yes – definitely",
			"You may rely on it.",
			"As I see it, yes.",
			"Most likely.",
			"Outlook good.",
			"Yes.",
			"Signs point to yes.",
			"Reply hazy, try again.",
			"Ask again later.",
			"Better not tell you now.",
			"Cannot predict now.",
			"Concentrate and ask again.",
			"Don't count on it.",
			"My reply is no.",
			"My sources say no.",
			"Outlook not so good.",
			"Very doubtful."
		};

		System.out.println("Here is my answer to your question: ");
		System.out.println(myQuestion);
		System.out.println(replies[replyIndex]);
  	}
}

File: python/Build_Python_Web_Apps_with_Flask/Introduction_to_Python/Control_Flow/Magic_8-Ball/magic8.py
# ----- Setting up ----- 
import random

name = "Csaba"
question = "Will I win the lottery?"
answer = ""

# -----  Generating a random number----- 
random_number = random.randint(1, 10)  # generate a random number between 1 (inclusive) and 9 (inclusive)
#print(random_number)

# ----- Control Flow ----- 
if random_number == 1:
  answer = "Yes - definitely."
elif random_number == 2:
  answer = "It is decidedly so."
elif random_number == 3:
  answer = "Without a doubt."
elif random_number == 4:
  answer = "Reply hazy, try again."
elif random_number == 5:
  answer = "Ask again later."
elif random_number == 6:
  answer = "Better not tell you now."
elif random_number == 7:
  answer = "My sources say no."
elif random_number == 8:
  answer = "Outlook not so good."
elif random_number == 9:
  answer = "Very doubtful."
elif random_number == 10:   # Optional Challenges
  answer = "What would you answer?"
else:
  answer = "Error"

# ----- Seeing the result -----
# question
if name != "":
  print(f"{name} asks: {question}")
else:
  print(f"Question: {question}")
 
# answer
if question != "":
  print(f"Magic 8-Ball's answer: {answer}")
else:
  print(f"The Magic 8-Ball cannot provide a fortune unless you ask it something.") 
  
